### PR TITLE
test: starship testing in ci

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
## Description

- Update `integration` GitHub workflow to mimic [multichain-testing](https://github.com/Agoric/agoric-sdk/blob/0ceb25cebf7ba6d8356ecf7fffe07eead623b7d1/.github/workflows/multichain-e2e.yml) workflow

## To Do Local

- [x] Start starship
    - `cd e2e-testing && make start`
- [x] Run override-chain-registry proposal
   - _happens as part of above command_
- [ ] orca-multichain.test.js
  - `yarn integration-test` from top-level
  - currently failing during core proposal `Cannot find file for internal module "./src/utils/start-helper.js"`

<details>
<summary>Start orca proposal chain logs</summary>

```log
2024-09-24T20:17:45.781Z SwingSet: vat: v1: installation orca settled; remaining: []
2024-09-24T20:17:45.919Z SwingSet: vat: v1: coreProposal: startOrcaContract
2024-09-24T20:17:45.919Z SwingSet: vat: v1: ----- OrCE.2  8 startOrcaContract()... 0.0.93
2024-09-24T20:17:45.920Z SwingSet: vat: v1: { consume: { agoricNames: Promise [Promise] {}, board: Promise [Promise] {}, chainStorage: Promise [Promise] {}, chainTimerService: Promise [Promise] {}, cosmosInterchainService: Promise [Promise] {}, localchain: Promise [Promise] {}, startUpgradable: Promise [Promise] {}, zoe: Promise [Promise] {} }, installation: { consume: { orca: Promise [Promise] {} }, produce: { orca: { resolve: [Function: resolve], reject: [Function: reject], reset: [Function: reset] } } }, instance: { produce: { orca: { resolve: [Function: resolve], reject: [Function: reject], reset: [Function: reset] } } } }
2024-09-24T20:17:45.920Z SwingSet: vat: v1: { options: undefined }
2024-09-24T20:17:45.920Z SwingSet: vat: v1: ----- OrCE.2  9 config { options: undefined }
2024-09-24T20:17:45.921Z SwingSet: vat: v1: ----- OrCE.2  10 permittedPowers { consume: { agoricNames: Promise [Promise] {}, board: Promise [Promise] {}, chainStorage: Promise [Promise] {}, chainTimerService: Promise [Promise] {}, cosmosInterchainService: Promise [Promise] {}, localchain: Promise [Promise] {}, startUpgradable: Promise [Promise] {}, zoe: Promise [Promise] {} }, installation: { consume: { orca: Promise [Promise] {} }, produce: { orca: { resolve: [Function: resolve], reject: [Function: reject], reset: [Function: reset] } } }, instance: { produce: { orca: { resolve: [Function: resolve], reject: [Function: reject], reset: [Function: reset] } } } }
2024-09-24T20:17:45.921Z SwingSet: vat: v1: ----- OrCE.2  11 produceInstance:
2024-09-24T20:17:45.921Z SwingSet: vat: v1: ----- OrCE.2  12 orcaInstallation Promise [Promise] {}
2024-09-24T20:17:45.921Z SwingSet: vat: v1: ----- OrCE.2  13 config is undefined, assigning installation to orcaInstallation
2024-09-24T20:17:45.922Z SwingSet: vat: v1: installation:
2024-09-24T20:17:45.922Z SwingSet: vat: v1: Object [Alleged: BundleIDInstallation] {}
2024-09-24T20:17:45.922Z SwingSet: vat: v1: permittedPowers
2024-09-24T20:17:45.923Z SwingSet: vat: v1: { consume: { agoricNames: Promise [Promise] {}, board: Promise [Promise] {}, chainStorage: Promise [Promise] {}, chainTimerService: Promise [Promise] {}, cosmosInterchainService: Promise [Promise] {}, localchain: Promise [Promise] {}, startUpgradable: Promise [Promise] {}, zoe: Promise [Promise] {} }, installation: { consume: { orca: Promise [Promise] {} }, produce: { orca: { resolve: [Function: resolve], reject: [Function: reject], reset: [Function: reset] } } }, instance: { produce: { orca: { resolve: [Function: resolve], reject: [Function: reject], reset: [Function: reset] } } } }
2024-09-24T20:17:45.923Z SwingSet: vat: v1: from inside startOrcaContract: { resolve: [Function: resolve], reject: [Function: reject], reset: [Function: reset] }
2024-09-24T20:17:45.923Z SwingSet: vat: v1: Promise [Promise] {}
2024-09-24T20:17:45.923Z SwingSet: vat: v1: Promise [Promise] {}
2024-09-24T20:17:45.923Z SwingSet: vat: v1: chainStorage
2024-09-24T20:17:45.923Z SwingSet: vat: v1: Promise [Promise] {}
2024-09-24T20:17:45.931Z SwingSet: vat: v1: Object [Alleged: ChainStorageNode] {}
2024-09-24T20:17:45.931Z SwingSet: vat: v1: DONE MAKING NODES v0.3
2024-09-24T20:17:45.936Z SwingSet: vat: v1: Object [Alleged: Board publishingMarshaller] {}
2024-09-24T20:17:45.937Z SwingSet: vat: v1: ----- OrCE.2  14 startOpts { label: 'orca', installation: Object [Alleged: BundleIDInstallation] {}, terms: undefined, privateArgs: { localchain: Object [Alleged: LocalChainKit public] {}, orchestrationService: Object [Alleged: Orchestration public] {}, storageNode: Object [Alleged: ChainStorageNode] {}, timerService: Object [Alleged: timerService] {}, agoricNames: Object [Alleged: NameHubKit nameHub] {}, marshaller: Object [Alleged: Board publishingMarshaller] {} } }
8:17PM INF Timed out dur=400 height=3263 module=consensus round=0 step=3
2024-09-24T20:17:48.110Z SwingSet: xsnap: v38: error during vat dispatch() of startVat,[object Object] (Error#1)
2024-09-24T20:17:48.110Z SwingSet: xsnap: v38: Error#1: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
2024-09-24T20:17:48.110Z SwingSet: xsnap: v38: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
 at apply ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:60)
 at execute (.../compartment-mapper/src/import-archive-lite.js:66)
 at execute (/bundled-source/.../node_modules/ses/src/module-instance.js:99)
 at imports (/bundled-source/.../node_modules/ses/src/module-instance.js:370)
 at (.../dapp-agoric-orca-contract/src/orca.contract.js:1)
 at execute (/bundled-source/.../node_modules/ses/src/module-instance.js:465)
 at compartmentImportNow (/bundled-source/.../node_modules/ses/src/compartment.js:64)
 at (/bundled-source/.../node_modules/ses/src/compartment.js:123)
 at ()

2024-09-24T20:17:48.125Z SwingSet: kernel: kernel terminating vat v38 (failure=true)
2024-09-24T20:17:48.129Z SwingSet: ls: v2: Logging sent error stack (Error#2)
2024-09-24T20:17:48.129Z SwingSet: ls: v2: Error#2: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
2024-09-24T20:17:48.129Z SwingSet: ls: v2: Error: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
 at apply ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:60)
 at newVatCallback (.../swingset-vat/src/vats/vat-admin/vat-vat-admin.js:523)
 at vatTerminated (.../swingset-vat/src/vats/vat-admin/vat-vat-admin.js:581)
 at apply ()
 at localApplyMethod (/bundled-source/.../node_modules/@endo/eventual-send/src/local.js:126)
 at apply ()
 at dispatchToHandler (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:159)
 at win (/bundled-source/.../node_modules/@endo/eventual-send/src/handled-promise.js:511)
 at ()

2024-09-24T20:17:48.129Z SwingSet: ls: v2: Error#2 ERROR_NOTE: Sent as error:liveSlots:v2#70001
2024-09-24T20:17:48.132Z SwingSet: ls: v9: Logging sent error stack (RemoteError(error:liveSlots:v2#70001)#1)
2024-09-24T20:17:48.132Z SwingSet: ls: v9: RemoteError(error:liveSlots:v2#70001)#1: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
2024-09-24T20:17:48.133Z SwingSet: ls: v9: Error: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
 at apply ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:60)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:351)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:309)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:437)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:398)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1064)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1088)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1358)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:59)
 at ()

2024-09-24T20:17:48.133Z SwingSet: ls: v9: RemoteError(error:liveSlots:v2#70001)#1 ERROR_NOTE: Sent as error:liveSlots:v9#70001
2024-09-24T20:17:48.136Z SwingSet: vat: v1: CORE_EVAL failed: (RemoteError(error:liveSlots:v9#70001)#1)
2024-09-24T20:17:48.136Z SwingSet: vat: v1: RemoteError(error:liveSlots:v9#70001)#1: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
2024-09-24T20:17:48.136Z SwingSet: vat: v1: Error: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
 at apply ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:60)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:351)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:309)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:437)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:398)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1064)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1088)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1358)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:59)
 at ()

2024-09-24T20:17:48.137Z SwingSet: xsnap: v1: RemoteError(error:liveSlots:v9#70001)#1 ERROR_NOTE: Sent as error:liveSlots:v1#70001
2024-09-24T20:17:48.137Z SwingSet: ls: v1: Logging sent error stack (RemoteError(error:liveSlots:v9#70001)#1)
2024-09-24T20:17:48.139Z SwingSet: xsnap: v10: UnhandledPromiseRejectionWarning: (RemoteError(error:liveSlots:v1#70001)#1)
2024-09-24T20:17:48.139Z SwingSet: xsnap: v10: RemoteError(error:liveSlots:v1#70001)#1: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
2024-09-24T20:17:48.139Z SwingSet: xsnap: v10: Error: Vat Creation Error: Error: Error: Cannot find file for internal module "./src/utils/start-helper.js" (with candidates "./src/utils/start-helper.js", "./src/utils/start-helper.js.js", "./src/utils/start-helper.js.json", "./src/utils/start-helper.js.node", "./src/utils/start-helper.js/index.js", "./src/utils/start-helper.js/index.json", "./src/utils/start-helper.js/index.node") in package file:///Users/0xpatrick/repos/dapp-orchestration-basics/contract/node_modules/@agoric/orchestration/
 at apply ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:60)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:351)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:309)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:437)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:398)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1064)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1088)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1358)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:59)
 at ()

8:17PM INF executed block height=3263 module=state num_invalid_txs=0 num_valid_txs=0
```
</details>

## To Do CI

The current `pr.yaml` are best estimate of what should work.

-  [x] We need `agoric` in path for `agoric run`, [1a2ecee](https://github.com/Agoric/dapp-orchestration-basics/pull/50/commits/1a2eceee8af5e027af63c4300de32800d55a3701)
- [x] override-chain-registry proposal passes
- [ ] orca bundle installs
    - see https://github.com/Agoric/dapp-orchestration-basics/pull/50#issuecomment-2372411166
- [ ] orca proposal passes
- [ ] orca integration test passes

   
   
## To Do Docs

- [ ] Update instructions to run starship from this repo instead of `agoric-sdk`
   - Follow up PR

### CODE REVIEW

File: e2e-testing/tools/sleep.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    operation: () => Promise<T>,
+    condition: (result: T) => boolean,
+    message: string,
+    options?: RetryOptions,
+  ) =>
+    retryUntilCondition(operation, condition, message, {
+      ...defaultOptions,
+      ...options,
+    });
+};
```
Comment (0xpatrickdev) [inline e2e-testing/tools/sleep.ts:77]: This is a dupe of `/contract/tools` we should DRY

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -24,35 +24,74 @@ jobs:
...
+        run: |
+          git clone https://github.com/Agoric/agoric-sdk.git
+          cd agoric-sdk
+          yarn install
+          yarn build
+          yarn link-cli
+          echo "$HOME/bin" >> $GITHUB_PATH
+
+      - name: Build dapp-agoric-basics
+        run: yarn build
```
Comment (0xpatrickdev) [inline .github/workflows/pr.yml:49]: Can remove, unless we are interested in testing the UI

File: e2e-testing/Makefile [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# see https://github.com/cosmology-tech/starship/blob/0e18757b8393357fc66426c5ee23da4ccf760e74/examples/getting-started/Makefile
```
Comment (0xpatrickdev) [inline e2e-testing/Makefile:1]:
```suggestion
# see https://github.com/Agoric/agoric-sdk/blob/472138d33b53fc2782fd415e992423571400e0dc/multichain-testing/config.yaml and https://github.com/cosmology-tech/starship/blob/0e18757b8393357fc66426c5ee23da4ccf760e74/examples/getting-started/Makefile

**Comment by 0xpatrickdev** [comment]
The orca bundle installation fails on this CI run: https://github.com/Agoric/dapp-orchestration-basics/actions/runs/11021918499/job/30610086193?pr=50. This might suggest the bundle is too large

File: .github/workflows/pr.yml (hunk 2 above)
Comment (Jovonni) [reply .github/workflows/pr.yml:49]: Do we plan on testing UI interactions in our pipeline, or just rely on `unit`?

**Comment by dckc** [comment]
> Is the bundle too large?

no:

```
  ✔ build-proposal › bundles small enough for Mainnet (7.7s)
    ℹ {
        bundleName: 'orca',
        compressedSize: '0.372 MB',
        originallySize: '1.900 MB',
      }
```
 -- [unit ci log](https://github.com/Agoric/dapp-orchestration-basics/actions/runs/11022310996/job/30611322397?pr=50#step:9:18)

**Comment by dckc** [comment]
btw... w.r.t. target branch: I wonder if the branch for #37 would be better. The idea behind `dc-u17-emery` / #47 was to take #37 and merge `main` in order to get the bundle size fix (#36).

**Comment by 0xpatrickdev** [comment]
On local with the latest deps, I now believe im running into:  https://github.com/Agoric/agoric-sdk/pull/10149

```log
agoric: run: running /../dapp-orchestration-basics/contract/test/builder/init-orca.js
  agoric: (TypeError#1)
  TypeError#1: Failed to load module "./src/orca.contract.js" in package "file:///../dapp-orchestration-basics/contract/" (1 underlying failures: Cannot find file for internal module "./src/types.js" (with candidates "./src/types.js", "./src/types.js.js", "./src/types.js.json", "./src/types.js.node", "./src/types.js/index.js", "./src/types.js/index.json", "./src/types.js/index.node") in package file:///../dapp-orchestration-basics/node_modules/@agoric/orchestration/
  ```

**Comment by 0xpatrickdev** [comment]
Closing in favor of #61. @dckc - thanks for taking this over the finish line!

**Pull request closed without merging**